test/vm.js

```diff
@@ -47,28 +47,30 @@ describe("revert opcode", function() {
     });
   });
 
-  it("should return a transaction receipt with status 0 on REVERT", function() {
-    var revertCode = testContext.revertContract.binary;
-    var revertAbi = JSON.parse(testContext.revertContract.abi);
+  it("should return a transaction receipt with status 0 on REVERT", async function() {
+    const revertCode = testContext.revertContract.binary;
+    const revertAbi = JSON.parse(testContext.revertContract.abi);
 
-    var RevertContract = new web3.eth.Contract(revertAbi);
+    const RevertContract = new web3.eth.Contract(revertAbi);
     RevertContract._code = revertCode;
-    return RevertContract.deploy({ data: revertCode })
-      .send({ from: testContext.accounts[0], gas: 3141592 })
-      .then(function(instance) {
-        // TODO: ugly workaround - not sure why this is necessary.
-        if (!instance._requestManager.provider) {
-          instance._requestManager.setProvider(web3.eth._provider);
-        }
-        return instance.methods.alwaysReverts(5).send({ from: testContext.accounts[0] });
-      })
-      .catch(function(err) {
-        assert.strictEqual(err.results[err.hashes[0]].error, "revert", "Expected error result not returned.");
-        return web3.eth.getTransactionReceipt(err.hashes[0]);
-      })
-      .then(function(receipt) {
-        assert.notStrictEqual(receipt, null, "Transaction receipt shouldn't be null");
-        assert.strictEqual(receipt.status, false, "Reverted (failed) transactions should have a status of FALSE.");
+
+    let receipt;
+    try {
+      const instance = await RevertContract.deploy({ data: revertCode }).send({
+        from: testContext.accounts[0],
+        gas: 3141592
       });
+
+      // TODO: ugly workaround - not sure why this is necessary.
+      if (!instance._requestManager.provider) {
+        instance._requestManager.setProvider(web3.eth._provider);
+      }
+      receipt = await instance.methods.alwaysReverts(5).send({ from: testContext.accounts[0] });
```

Comment:
Update to `await assert.rejects` pattern.

Reply:
It turns out that we need to grab the transaction hash from the error object in order to grab the receipt.